operator/fixup_structs/main.go

```diff
@@ -126,7 +126,7 @@ func main() {
 			tmp = append(tmp, lines[i+1])
 			i += 2
 		case strings.Contains(l, goFieldJSONTagPrefix) && strings.Contains(l, "_"):
-			rgx := regexp.MustCompile(`json\:"[a-z|0-9]+_[a-z|0-9]+,`)
+			rgx := regexp.MustCompile(`json\:"[a-z|0-9|_]+,`)
```

Comment:
I also encountered this issue. The old one doesn't match string begins with `_`, how about:
```suggestion
			rgx := regexp.MustCompile(`json\:"[a-z|0-9]+(_[a-z|0-9]+)+,`)
```

Reply:
I'm not sure I follow your suggestion. Could you provide an example of where this would work and the one in the PR would fail?

Reply:
The old one's problem is it can't match strings with more than one "_", e.g "xxx_yyy_zzz", it should be fixed to work with them; the one in this PR matches "xxx_yyy_zzz", but it also matches strings begins with "_", e.g. "_xxx". 

And the old one doesn't match strings without "_" in it, e.g "xxxx"; but the one in the PR matches strings without "_", that's, the replacing function is always called.

Currently I guess there are no fields begin with "_", so the one in the PR doesn't cause any issue, but it's not necessary to extend the matching space.

Reply:
Generally fields starting with _ are not allowed, at least these are caught by the linter since the spec is not explicit about this. But to be extra safe I've put in a check that aborts the tool if such a tag name is found.